### ISSUE TITLE
Web Worker Example

### ISSUE THREAD

**Issue by siddjain** [Microsoft/TypeScript#5676]
Is there any place I can find an example of writing a web worker in typescript? I have created a new file myworker.ts that looks like below:

```ts
/// <reference path="IWorkerMessages.ts" />
self.onmessage = (ev: MessageEvent) =>
{
    var data = <IWorkerIncoming>ev.data;
    

} 

// I get a compiler error while trying to define self.deserialize
self.deserialize = (data: IWorkerIncoming) =>
{

}
```

could the file be refactored to something like
```
class MyWorker implements Worker
{
}
```

*Edited by @DanielRosenwasser to add code fences.*

**Comment by DanielRosenwasser** [comment]
Webworkers in TypeScript work the same as in JavaScript. The only thing in question is where to get the typings from. Take a look at http://blog.scottlogic.com/2015/08/10/ts-vs-proj.html and #4855.

In the future, StackOverflow is a better venue for these types of questions, just as a heads up.

**Comment by siddjain** [comment]
Thanks for the link. Actually I found it more confusing than helpful to be honest. I am not using any virtual projects. Further I am developing a single page app with no MVC. I do my compilation within VS and am not using command line. The problems I am struggling with are:
- How to compile the web worker code into a separate js
- Looking at examples of web workers, it seems the compiled js ought to look something like:
self.onmessage = function()...
but if I write a typescript file that implements the Worker interface, that is not the compiled js I will get.
I posted here as I thought response would be faster (and it was, thanks :)
